File: clutch-app-android/app/src/main/java/com/clutch/app/data/model/MaintenanceService.kt
package com.clutch.app.data.model

import com.google.gson.annotations.SerializedName

data class MaintenanceService(
    @SerializedName("_id")
    val id: String,
    @SerializedName("serviceGroup")
    val serviceGroup: String,
    @SerializedName("serviceName")
    val serviceName: String,
    @SerializedName("description")
    val description: String? = null,
    @SerializedName("icon")
    val icon: String? = null,
    @SerializedName("isActive")
    val isActive: Boolean = true
)

data class MaintenanceServiceGroup(
    val groupName: String,
    val services: List<MaintenanceService>
)

File: shared-backend/models/MaintenanceService.js
const mongoose = require('mongoose');

const maintenanceServiceSchema = new mongoose.Schema({
  serviceGroup: {
    type: String,
    required: true,
    trim: true
  },
  serviceName: {
    type: String,
    required: true,
    trim: true
  },
  description: {
    type: String,
    default: null
  },
  icon: {
    type: String,
    default: null
  },
  isActive: {
    type: Boolean,
    default: true
  }
}, {
  timestamps: true
});

// Index for efficient queries
maintenanceServiceSchema.index({ serviceGroup: 1, isActive: 1 });

module.exports = mongoose.model('MaintenanceService', maintenanceServiceSchema);
